src/OpenTelemetry.Exporter.LightStep/Implementation/TraceExporterHandler.cs

```diff
@@ -0,0 +1,134 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.LightStep.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+    using OpenTelemetry.Trace;
+    using OpenTelemetry.Trace.Export;
+
+    public class TraceExporterHandler : IHandler
+    {
+        private readonly LightStepTraceExporterOptions options;
+        private readonly HttpClient httpClient;
+
+        public TraceExporterHandler(LightStepTraceExporterOptions options, HttpClient client)
+        {
+            this.options = options;
+            this.httpClient = client ?? new HttpClient();
+            this.httpClient.Timeout = this.options.SatelliteTimeout;
+        }
+
+        public async Task ExportAsync(IEnumerable<SpanData> spanDataList)
+        {
+            var lsReport = new LightStepReport
+            {
+                Auth = new Authentication { AccessToken = this.options.AccessToken },
+                Reporter = new Reporter
+                {
+                    // TODO: ReporterID should be randomly generated.
+                    ReporterId = 219,
+                    Tags = new List<Tag>
+                    {
+                        new Tag { Key = "lightstep.component_name", StringValue = this.options.ServiceName },
+                    },
+                },
+            };
+
+            foreach (var data in spanDataList)
+            {
+                lsReport.Spans.Add(this.GenerateSpan(data));
+            }
+
+            try
+            {
+                await this.SendSpansAsync(lsReport);
+            }
+            catch (Exception)
+            {
+                // ignore
+            }
+        }
+
+        private LightStepSpan GenerateSpan(SpanData data)
+        {
+            var duration = data.EndTimestamp - data.StartTimestamp;
+            var span = new LightStepSpan();
+
+            if (data.ParentSpanId != default)
+            {
+                var sc = new SpanContext { SpanId = Convert.ToUInt64(data.ParentSpanId.ToHexString(), 16) };
+                span.References.Add(new Reference
+                {
+                    Relationship = "CHILD_OF", SpanContext = sc,
+                });
+            }
+
+            span.OperationName = data.Name;
+            var sid = Convert.ToUInt64(data.Context.SpanId.ToHexString(), 16);
+
+            var longTraceId = data.Context.TraceId.ToHexString()
```

Comment:
`ToHexString` is a string allocation, please cache it
Also TraceId is a Span<byte> you can slice it without any allocations and after you have a slice that you need, convert it to string with BitConverter

Reply:
Span/Slice doesn't seem to be available in NetStandard2? I can intern the strings though.

Reply:
Spans live in System.Memory nuget that is available on net fx  or netstandard2